### PR TITLE
Add reorder tautomers function and accompanying tests

### PR DESCRIPTION
**Pull request by chriswzou**

### CODE REVIEW

File: rdkit/Chem/MolStandardize/UnitTestMolStandardize.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import unittest
+
+from rdkit import Chem
+from rdkit.Chem import MolStandardize
+from rdkit.Chem.MolStandardize import rdMolStandardize
+
+class TestCase(unittest.TestCase):
+
+    def testBasic(self):
+        print(MolStandardize.__file__)
```
Comment (greglandrum) [inline rdkit/Chem/MolStandardize/UnitTestMolStandardize.py]: Please remove these extra print statements from the tests.

File: rdkit/Chem/MolStandardize/UnitTestMolStandardize.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        print(Chem.__file__)
+        print(rdMolStandardize.__file__)
+        m = Chem.MolFromSmiles('Oc1c(cccc3)c3nc2ccncc12')
+        enumerator = rdMolStandardize.TautomerEnumerator()
+        canon = enumerator.Canonicalize(m)
+        reord = MolStandardize.ReorderTautomers(m)[0]
+        canonSmile = Chem.MolToSmiles(canon)
+        reordSmile = Chem.MolToSmiles(reord)
+        self.assertEquals(canonSmile, reordSmile)
```
Comment (greglandrum) [inline rdkit/Chem/MolStandardize/UnitTestMolStandardize.py]: This is a good test that the first value in the list is the canonical tautomer. I would suggest also checking the length of the list returned by `ReorderTautomers()`

File: rdkit/Chem/MolStandardize/__init__.py [hunk 3]
```diff
@@ -11,21 +11,40 @@
...
 log.addHandler(logging.NullHandler())
+
+
+def ReorderTautomers(molecule):
+    """Returns the list of the molecule's tautomers
+    so that the canonical one as determined by the canonical
+    scoring system in TautomerCanonicalizer appears first.
+
+    :param molecule: An RDKit Molecule object.
+    :return: An array of Molecule objects.
```
Comment (greglandrum) [inline rdkit/Chem/MolStandardize/__init__.py]:
```suggestion
    :return: A list of Molecule objects.
```

File: rdkit/Chem/MolStandardize/tautomer.py [hunk 4]
```diff
@@ -9,11 +9,11 @@
 :license: MIT, see LICENSE file for more details.
```
Comment (greglandrum) [inline rdkit/Chem/MolStandardize/tautomer.py]: I don't think you need to make any edits at all to this file for this PR

Comment (chriswzou) [reply rdkit/Chem/MolStandardize/tautomer.py]: I've manually put the changes back. To be honest, not sure how they came about—suspect it was Pycharm? Will keep an eye out for any like these in the future.

File: .gitignore [hunk 5]
```diff
@@ -99,3 +99,5 @@ __pycache__/
 /rdkit/ML/Data/test_data/testgeneral.dat.pkl
 /rdkit/ML/Data/test_data/testquant.qdat.pkl
 .DS_Store
+.idea
```
Comment (greglandrum) [inline .gitignore]: Please don't add files to the hosted `.gitignore` that are specific to your system.
These changes aren't related to the PR you're making

File: rdkit/Chem/MolStandardize/UnitTestMolStandardize.py [hunk 6]
```diff
@@ -24,5 +21,14 @@ def testBasic(self):
         reordSmile = Chem.MolToSmiles(reord)
         self.assertEquals(canonSmile, reordSmile)
 
+    def testLength(self):
+        m = Chem.MolFromSmiles('Oc1c(cccc3)c3nc2ccncc12')
+        enumerator = rdMolStandardize.TautomerEnumerator()
+        tauts = enumerator.Enumerate(m)
+        reordtauts = MolStandardize.ReorderTautomers(m)
+        self.assertEquals(len(reordtauts), len(tauts))
```
Comment (greglandrum) [inline rdkit/Chem/MolStandardize/UnitTestMolStandardize.py]: nice. That's a good test

File: .gitignore (hunk 5 above)
Comment (chriswzou) [reply .gitignore]: Note to self: don't use StackOverflow solutions without thinking about them first! Of course, .gitignore is as much a part of the repository as anything else. I'll look through my IDE settings and see if there's a way to not include the .idea files from there instead.

**Comment by greglandrum** [comment]
@chriswzou : thanks for this! I will merge it after the CI builds pass

**Pull request merged**